### DISCUSSION TITLE
Would be perfect with a  "choose resolution" option

### DISCUSSION THREAD

**Discussion by dtpitch** [MonitorControl/MonitorControl#1055] [Ideas]
Hi,
The app works fine with my LG C1 48' except for :
- The brightness fades only when the bar gets under 50%
- Using the contrast option make the mac crash with grey "snow" on all the screen

BTW I would love if I could choose between 4k or 1440p just clicking on the app icon and have that option.

Thx for your work !

**Comment by waydabber** [comment]
Hi there, take a look at this if you want a resolution slider: https://github.com/waydabber/BetterDummy/releases

**Comment by the0neyouseek** [comment]
If the brightness drop only after 50% it means ddc isn't working (could be the monitor or how you're plugged to it). You can uncheck the "use hardware ddc control" checkbox in the display setting tab.

**Comment by Apolsus** [comment]
I noticed that this software, now called BetterDisplay, provides almost repetitive functionality as the MonitorControl. How do these two software compare? I don't want to install two software with repetitive functions.

And I notice that you are the developer of these two software. Do you know how to migrate the resolution technology?

**Reply by waydabber** [reply]
If you only want brightness control and nothing fancy (like extra XDR/HDR brightness, additional DDC and color controls, resolution related stuff etc), then you do well if you use MonitorControl as it is more focused on a specific app.

BetterDisplay covers a whole lot more (even the free version). Some history - the original scope of BD was just about resolutions, then brightness control was also added due to the fact that this is integral to the XDR/HDR extra brightness function which I wanted to implement somewhere - I could not opensource that and add it to MC since I co-developed it with an other guy who (understandably) did not really like the idea of making it opensource or free. Then as a consequence I also started adding more and more MC features to BD since BD and MC could not run side-by-side without clash in functionalities and I wanted to give a way for MC users to use BD as well (for a while there was a compatibility toggle in BC to disable all things that would clash with MC though but it became untenable after a while) + I also wanted to reengineer and greatly expand the entire control thing that was in MC to be more generic and to allow extra functionality (so BD is a complete reimplementation internally using a new approach, not a copy-paste of MC). An other consideration was that MC was already somewhat "overdeveloped" (just compare it to version 2.0) into a state when it seemed like other contributors could not contribute much to it - the maintainer board felt this to be counter-productive and contrary to the original simpler vision of the app (rightfully) - adding even more complexity and features in the earlier pace or a complete-rewrite was just not something felt appropriate in that light so I shifted focus.

All in all, both apps are great, I invested a lot of time and energy into both, use whichever you love more. Important to consider that while BetterDisplay has much more features and is in a lot of respects much more up to date, it is not open source - so if open-source is important to you, use have no choice but to use MonitorControl. :)

You can compare the two apps by going to the [BetterDisplay's github page](https://github.com/waydabber/BetterDisplay#readme) and opening the `For a more detailed list and comparison with some popular alternatives, click here...` section. Here you'll also see what is free and what is paid in BetterDisplay.

Hope this info helps!

**Reply by waydabber** [reply]
Just one clarification: the above does not mean I abandoned MC, I still plan to bring some improvements over (most notably M2 support or M1 HDMI support, but there are a lot of things I know now should be done differently than how it is done in MC). But the two apps really diverged internally so it's not that that simple and if I start working on MC, I am afraid it will be like a thread which I pull and the whole thing unravels - meaning I will feel compelled to rewrite larger parts of the app. I plan to put some effort into it maybe during summer (an other idea is to do an opposite thing - create a stripped down, open-source version of BD which should give the opportunity for any contribution with the right skillset to carry anything over to MC). I also hope (and encourage) that meanwhile somebody (anybody) would come and contribute to MC making things happen.

Regarding M2 and M1 HDMI support I have some other reservations as well (to be honest, even though I know it's probably not an elegant thing to voice) - these features are totally free but closed source in BD and by bringing them to MC, these would be also open source. However I worked on these stuff hard. It seems like there is an other player who tends to just copy-paste these things and repackage it and sell it for a lot of money (more actually than BD with all its features cost) in a generally inferior product. This is actually something that annoys me (I know it should not, but still, I can't do much about it), so this makes me hesitant. An other important thing to know is that original DDC support for M1 is not my original work (even though I implemented it in MC), it was branched off from the early work of @alin23, the creator of Lunar who actually silently contributed a lot to MC by providing the initial recipe for doing DDC on Apple Silicon (he is such a good soul!) and helped me out in the early days with a lot of things when I was just getting to know this stuff (but I think the relationship was beneficial, I also gave some important input to make Lunar better, I think he'd agree with that). Now I know for fact that what I just mentioned regarding that third player annoys @alin23 very much as well (this is not a secret as they did had some argument over it in some forum somewhere), and I feel it would not be fair to him to rush opensourcing M2 support and then watching that other player come out with his version 2 days later (note: @alin23 also figured out M2 and M1 HDMI support independently of me so Lunar has it now as well, but partly due to this issue, he - understandably - did not open-source that this time). So I guess the best approach is just to wait until somebody else also does the trick independently and open-sources it and simply do nothing until that time. All this is also mixed with the fact that - probably understandably (? :)) - I am generally proud of BD and don't really mind that (given the circumstances) users are therefore using it if they want M1 HDMI or M2 support for DDC - this might be some conflict of interests on my part (as both one of the maintainers of MC and the creator of BD). I think the general public will not suffer though with this as those who want M1 HDMI or M2 support for DDC can get it fully functional and fully free by downloading BD, without paying a dime.
